pkl/private/pkl_codegen_java.bzl

```diff
@@ -140,7 +141,8 @@ def pkl_java_library(name, srcs, module_path = [], generate_getters = None, deps
         tags = tags,
     )
 
-    pkl_deps = [Label("//pkl/private:pkl-tools")]
+    if len(pkl_deps) == 0:
+        pkl_deps = [Label("//pkl/private:pkl-config-java-internal")]
```

Comment:
That behavior can be surprising for the user if they add a java dependency that is not directly related to the pkl-config-java. Would that ever be a thing?

If not, maybe the attribute can reflect that more specificly (`pkl_config_java_deps` or similar).
If it might be a thing, I'm not sure what to suggest.